integration_tests/it_exemplars_micrometer_tracing/src/main/java/io/prometheus/client/it/exemplars_micrometer_tracing/ExampleApplication.java

```diff
@@ -0,0 +1,53 @@
+package io.prometheus.client.it.exemplars_micrometer_tracing;
+
+import io.micrometer.tracing.Span;
+import io.micrometer.tracing.Tracer;
+import io.micrometer.tracing.test.simple.SimpleSpan;
+import io.micrometer.tracing.test.simple.SimpleTracer;
+import io.prometheus.client.Counter;
+import io.prometheus.client.exemplars.DefaultExemplarSampler;
+import io.prometheus.client.exemplars.ExemplarConfig;
+import io.prometheus.client.exemplars.ExemplarSampler;
+import io.prometheus.client.exemplars.tracer.micrometer.tracing.MicrometerTracingSpanContextSupplier;
+import io.prometheus.client.exporter.HTTPServer;
+
+import java.io.IOException;
+
+/**
+ * Example application using Micrometer Tracing.
+ */
+public class ExampleApplication {
+
+  public static void main(String[] args) throws IOException, InterruptedException {
+    SimpleTracer tracer = new SimpleTracer();
+    ExemplarSampler exemplarSampler = new DefaultExemplarSampler(new MicrometerTracingSpanContextSupplier(tracer));
+    ExemplarConfig.setCounterExemplarSampler(exemplarSampler);
+    ExemplarConfig.setHistogramExemplarSampler(exemplarSampler);
+    ExemplarConfig.enableExemplars();
```

Comment:
It should not be necessary to set the `ExemplarSampler` explicitly. The idea is that `client_java` supports Exemplars automatically out-of-the-box if a supported tracer is found.

The magic happens in the `Tracer.findSpanContextSupplier()` in the `simpleclient` module. It checks if `OpenTelemetrySpanContextSupplier.isAvailable()`, then if `OpenTelemetryAgentSpanContextSupplier.isAvailable()`.

Please copy-and-paste these lines and add a check if `MicrometerTracingSpanContextSupplier.isAvailable()`.

As you are the first contributor adding Exemplars support, please provide feedback. Here's some context on the current implementation. Please let me know if this makes sense, or if there is a better or simpler way to do it:

Requirements:

1. Exemplars work out-of-the-box if a supported tracer is detected. For example, if you attach the OpenTelemetry Java agent to an application using `client_java`, you will automatically get Exemplars without changing your code. In the same way, if you add Micrometer tracing to an application using `client_java` you should automatically get Exemplars without any code change.
2. It should not be possible to break an application by adding `simpleclient` as a dependency.

Achieving (2) was easy as long as `simpleclient` was just a metrics library without dependencies. However, to implement Exemplar support we had to add dependencies like `opentelemetry-api`. This can be problematic if an application ships with another version of `opentelemetry-api`.

We took two steps to avoid potential issues:
* The `opentelemetry-api` dependency is `provided` to make sure we use the tracer that ships with the application at runtime.
* If there are incompatible API changes between different `opentelemetry-api` versions there might be runtime errors when using different versions at runtime and at compile time. As a last resort, we allow users to exclude the `simpleclient_tracer_otel` dependency from `simpleclient` to get rid of OTel exemplars for good. That's why we catch the `NoClassDefFoundError` in `findSpanContextSupplier()`. There's even an integration test removing the dependency and making sure that `simpleclient` still works.

Anyway, if you have any thoughts on this let me know.